Firestore/core/src/firebase/firestore/core/firestore_client.h

```diff
@@ -86,15 +86,16 @@ class FirestoreClient : public std::enable_shared_from_this<FirestoreClient> {
 
   /**
    * Terminates this client, cancels all writes / listeners, and releases all
-   * resources.
+   * resources. The client enters a restricted mode where only a few methods
+   * like ClearPersistence are allowed.
    */
   void TerminateAsync(util::StatusCallback callback);
```

Comment:
FWIW, to me, `Terminate` seems like a stronger term than `Stop`, so I find it a little surprising that `Stop` actually does more than `Terminate`. What do you think?